ethcore/src/client/test_client.rs

```diff
@@ -607,12 +607,12 @@ impl EngineInfo for TestBlockChainClient {
 }
 
 impl BlockChainClient for TestBlockChainClient {
-	fn replay(&self, _id: TransactionId, _analytics: CallAnalytics) -> Result<Executed, CallError> {
-		self.execution_result.read().clone().unwrap()
+	fn replay(&self, _id: TransactionId, _analytics: CallAnalytics) -> Result<(Option<H256>,Executed), CallError> {
+		Ok((None,self.execution_result.read().clone().unwrap()?))
 	}
 
-	fn replay_block_transactions(&self, _block: BlockId, _analytics: CallAnalytics) -> Result<Box<Iterator<Item = Executed>>, CallError> {
-		Ok(Box::new(self.execution_result.read().clone().unwrap().into_iter()))
+	fn replay_block_transactions(&self, _block: BlockId, _analytics: CallAnalytics) -> Result<Box<Iterator<Item = (Option<H256>,Executed)>>, CallError> {
+		Ok(Box::new(None.into_iter().zip(self.execution_result.read().clone().unwrap().into_iter())))
```

Comment:
Same as above. Block info are available in TestBlockChainClient, so I think it may be better to fetch that available info.